Issues with loading the images from Category

When the category is selected from the dropdown and moved to the next window its showing an error 'Unable to parse preview'. And if we clicked the ok button and on next page it will be showing the memory usage information but the images are not getting loaded even though the category has very few images too..

- OS: Windows
- Commons extension Version: 0.1.3
- OpenRefine Version 3.9.3 / 3.9.5

I tried it too. The whole data set can't be imported/created. First error message is that no preview is possible. If you try to create the project anyway it tries to load the data forever. So I stoped the process.

From the OR code it seems to me, that it might be the same login problem we had with wikibase in OR. [Wikidata blocking OpenRefine because it doesn't set User-Agent string #7383](https://github.com/OpenRefine/OpenRefine/issues/7383)
What do you think @Abbe98 ?

Same issue with the extension is also discussed in the [OpenRefine Forum](https://forum.openrefine.org/t/issue-with-openrefine-wikimedia-commons-extension/2531/2)

That is the code OR generated befor I stoped the process:
`09:56:44.322 [..monsImportingController] doPost::CommonsServiceException::{} (656ms)
java.io.UncheckedIOException: com.fasterxml.jackson.core.JsonParseException: Unrecognized token 'Please': was expecting (JSON String, Number, Array, Object or token 'null', 'true' or 'false')
 at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled); line: 1, column: 7]
        at org.openrefine.extensions.commons.importer.FileFetcher.<init>(FileFetcher.java:36)
        at org.openrefine.extensions.commons.importer.FileFetcher.fetchCategoryMembers(FileFetcher.java:88)
        at org.openrefine.extensions.commons.importer.FileFetcher.fetchDirectFileMembers(FileFetcher.java:99)
        at org.openrefine.extensions.commons.importer.FileFetcher.listCategoryMembers(FileFetcher.java:127)
        at org.openrefine.extensions.commons.importer.CommonsImporter.parse(CommonsImporter.java:107)
        at org.openrefine.extensions.commons.importer.CommonsImporter.parsePreview(CommonsImporter.java:45)
        at org.openrefine.extensions.commons.importer.CommonsImportingController.doParsePreview(CommonsImportingController.java:148)
        at org.openrefine.extensions.commons.importer.CommonsImportingController.doPost(CommonsImportingController.java:79)
        at com.google.refine.commands.importing.ImportingControllerCommand.doPost(ImportingControllerCommand.java:69)
        at com.google.refine.RefineServlet.service(RefineServlet.java:187)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:750)
        at org.eclipse.jetty.servlet.ServletHolder$NotAsync.service(ServletHolder.java:1410)
        at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:764)
        at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:529)
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:131)
        at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:578)
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:122)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:223)
        at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1570)
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:131)
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:122)
        at org.eclipse.jetty.server.handler.gzip.GzipHandler.handle(GzipHandler.java:790)
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:122)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:223)
        at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1384)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:176)
        at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:484)
        at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1543)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:174)
        at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1306)
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:129)
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:122)
        at com.google.refine.ValidateHostHandler.handle(ValidateHostHandler.java:93)
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:122)
        at org.eclipse.jetty.server.Server.handle(Server.java:563)
        at org.eclipse.jetty.server.HttpChannel$RequestDispatchable.dispatch(HttpChannel.java:1598)
        at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:753)
        at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:501)
        at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:282)
        at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:314)
        at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:100)
        at org.eclipse.jetty.io.SelectableChannelEndPoint$1.run(SelectableChannelEndPoint.java:53)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.base/java.lang.Thread.run(Unknown Source)
Caused by: com.fasterxml.jackson.core.JsonParseException: Unrecognized token 'Please': was expecting (JSON String, Number, Array, Object or token 'null', 'true' or 'false')
 at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled); line: 1, column: 7]
        at com.fasterxml.jackson.core.JsonParser._constructError(JsonParser.java:2584)
        at com.fasterxml.jackson.core.JsonParser._constructReadException(JsonParser.java:2610)
        at com.fasterxml.jackson.core.JsonParser._constructReadException(JsonParser.java:2618)
        at com.fasterxml.jackson.core.base.ParserMinimalBase._reportError(ParserMinimalBase.java:825)
        at com.fasterxml.jackson.core.json.ReaderBasedJsonParser._reportInvalidToken(ReaderBasedJsonParser.java:3017)
        at com.fasterxml.jackson.core.json.ReaderBasedJsonParser._handleOddValue(ReaderBasedJsonParser.java:2051)
        at com.fasterxml.jackson.core.json.ReaderBasedJsonParser.nextToken(ReaderBasedJsonParser.java:780)
        at com.fasterxml.jackson.databind.ObjectMapper._readTreeAndClose(ObjectMapper.java:4946)
        at com.fasterxml.jackson.databind.ObjectMapper.readTree(ObjectMapper.java:3281)
        at org.openrefine.extensions.commons.importer.FileFetcher.getCallResults(FileFetcher.java:57)
        at org.openrefine.extensions.commons.importer.FileFetcher.<init>(FileFetcher.java:34)
        ... 44 more
09:57:18.650 [                   refine] POST /command/core/importing-controller (34327ms)
Exception in thread "Thread-1" java.io.UncheckedIOException: com.fasterxml.jackson.core.JsonParseException: Unrecognized token 'Please': was expecting (JSON String, Number, Array, Object or token 'null', 'true' or 'false')
 at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled); line: 1, column: 7]
        at org.openrefine.extensions.commons.importer.FileFetcher.<init>(FileFetcher.java:36)
        at org.openrefine.extensions.commons.importer.FileFetcher.fetchCategoryMembers(FileFetcher.java:88)
        at org.openrefine.extensions.commons.importer.FileFetcher.fetchDirectFileMembers(FileFetcher.java:99)
        at org.openrefine.extensions.commons.importer.FileFetcher.listCategoryMembers(FileFetcher.java:127)
        at org.openrefine.extensions.commons.importer.CommonsImporter.parse(CommonsImporter.java:107)
        at org.openrefine.extensions.commons.importer.CommonsImportingController$1.run(CommonsImportingController.java:212)
Caused by: com.fasterxml.jackson.core.JsonParseException: Unrecognized token 'Please': was expecting (JSON String, Number, Array, Object or token 'null', 'true' or 'false')
 at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled); line: 1, column: 7]
        at com.fasterxml.jackson.core.JsonParser._constructError(JsonParser.java:2584)
        at com.fasterxml.jackson.core.JsonParser._constructReadException(JsonParser.java:2610)
        at com.fasterxml.jackson.core.JsonParser._constructReadException(JsonParser.java:2618)
        at com.fasterxml.jackson.core.base.ParserMinimalBase._reportError(ParserMinimalBase.java:825)
        at com.fasterxml.jackson.core.json.ReaderBasedJsonParser._reportInvalidToken(ReaderBasedJsonParser.java:3017)
        at com.fasterxml.jackson.core.json.ReaderBasedJsonParser._handleOddValue(ReaderBasedJsonParser.java:2051)
        at com.fasterxml.jackson.core.json.ReaderBasedJsonParser.nextToken(ReaderBasedJsonParser.java:780)
        at com.fasterxml.jackson.databind.ObjectMapper._readTreeAndClose(ObjectMapper.java:4946)
        at com.fasterxml.jackson.databind.ObjectMapper.readTree(ObjectMapper.java:3281)
        at org.openrefine.extensions.commons.importer.FileFetcher.getCallResults(FileFetcher.java:57)
        at org.openrefine.extensions.commons.importer.FileFetcher.<init>(FileFetcher.java:34)
        ... 5 more
09:57:19.013 [                   refine] POST /command/core/importing-controller (363ms)
09:57:19.113 [..monsImportingController] doPost::CommonsServiceException::{} (100ms)
java.io.UncheckedIOException: com.fasterxml.jackson.core.JsonParseException: Unrecognized token 'Please': was expecting (JSON String, Number, Array, Object or token 'null', 'true' or 'false')
 at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled); line: 1, column: 7]
        at org.openrefine.extensions.commons.importer.FileFetcher.<init>(FileFetcher.java:36)
        at org.openrefine.extensions.commons.importer.FileFetcher.fetchCategoryMembers(FileFetcher.java:88)
        at org.openrefine.extensions.commons.importer.FileFetcher.fetchDirectFileMembers(FileFetcher.java:99)
        at org.openrefine.extensions.commons.importer.FileFetcher.listCategoryMembers(FileFetcher.java:127)
        at org.openrefine.extensions.commons.importer.CommonsImporter.parse(CommonsImporter.java:107)
        at org.openrefine.extensions.commons.importer.CommonsImporter.parsePreview(CommonsImporter.java:45)
        at org.openrefine.extensions.commons.importer.CommonsImportingController.doParsePreview(CommonsImportingController.java:148)
        at org.openrefine.extensions.commons.importer.CommonsImportingController.doPost(CommonsImportingController.java:79)
        at com.google.refine.commands.importing.ImportingControllerCommand.doPost(ImportingControllerCommand.java:69)
        at com.google.refine.RefineServlet.service(RefineServlet.java:187)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:750)
        at org.eclipse.jetty.servlet.ServletHolder$NotAsync.service(ServletHolder.java:1410)
        at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:764)
        at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:529)
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:131)
        at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:578)
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:122)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:223)
        at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1570)
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:131)
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:122)
        at org.eclipse.jetty.server.handler.gzip.GzipHandler.handle(GzipHandler.java:790)
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:122)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:223)
        at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1384)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:176)
        at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:484)
        at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1543)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:174)
        at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1306)
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:129)
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:122)
        at com.google.refine.ValidateHostHandler.handle(ValidateHostHandler.java:93)
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:122)
        at org.eclipse.jetty.server.Server.handle(Server.java:563)
        at org.eclipse.jetty.server.HttpChannel$RequestDispatchable.dispatch(HttpChannel.java:1598)
        at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:753)
        at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:501)
        at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:282)
        at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:314)
        at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:100)
        at org.eclipse.jetty.io.SelectableChannelEndPoint$1.run(SelectableChannelEndPoint.java:53)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.base/java.lang.Thread.run(Unknown Source)
Caused by: com.fasterxml.jackson.core.JsonParseException: Unrecognized token 'Please': was expecting (JSON String, Number, Array, Object or token 'null', 'true' or 'false')
 at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled); line: 1, column: 7]
        at com.fasterxml.jackson.core.JsonParser._constructError(JsonParser.java:2584)
        at com.fasterxml.jackson.core.JsonParser._constructReadException(JsonParser.java:2610)
        at com.fasterxml.jackson.core.JsonParser._constructReadException(JsonParser.java:2618)
        at com.fasterxml.jackson.core.base.ParserMinimalBase._reportError(ParserMinimalBase.java:825)
        at com.fasterxml.jackson.core.json.ReaderBasedJsonParser._reportInvalidToken(ReaderBasedJsonParser.java:3017)
        at com.fasterxml.jackson.core.json.ReaderBasedJsonParser._handleOddValue(ReaderBasedJsonParser.java:2051)
        at com.fasterxml.jackson.core.json.ReaderBasedJsonParser.nextToken(ReaderBasedJsonParser.java:780)
        at com.fasterxml.jackson.databind.ObjectMapper._readTreeAndClose(ObjectMapper.java:4946)
        at com.fasterxml.jackson.databind.ObjectMapper.readTree(ObjectMapper.java:3281)
        at org.openrefine.extensions.commons.importer.FileFetcher.getCallResults(FileFetcher.java:57)
        at org.openrefine.extensions.commons.importer.FileFetcher.<init>(FileFetcher.java:34)
        ... 44 more
09:57:22.542 [                   refine] POST /command/core/get-importing-job-status (3428ms)
09:57:23.548 [                   refine] POST /command/core/get-importing-job-status (1006ms)
09:57:24.550 [                   refine] POST /command/core/get-importing-job-status (1002ms)
09:57:25.553 [                   refine] POST /command/core/get-importing-job-status (1002ms)
09:57:26.555 [                   refine] POST /command/core/get-importing-job-status (1002ms)
09:57:27.563 [                   refine] POST /command/core/get-importing-job-status (1007ms)
09:57:28.696 [                   refine] POST /command/core/get-importing-job-status (1133ms)
09:57:29.699 [                   refine] POST /command/core/get-importing-job-status (1002ms)
09:57:30.716 [                   refine] POST /command/core/get-importing-job-status (1016ms)
09:57:31.718 [                   refine] POST /command/core/get-importing-job-status (1002ms)
09:57:32.720 [                   refine] POST /command/core/get-importing-job-status (1002ms)
09:57:33.723 [                   refine] POST /command/core/get-importing-job-status (1002ms)
09:57:34.726 [                   refine] POST /command/core/get-importing-job-status (1002ms)
09:57:35.828 [                   refine] POST /command/core/get-importing-job-status (1102ms)
09:57:36.831 [                   refine] POST /command/core/get-importing-job-status (1002ms)
09:57:37.833 [                   refine] POST /command/core/get-importing-job-status (1002ms)
09:57:38.836 [                   refine] POST /command/core/get-importing-job-status (1002ms)
09:57:39.838 [                   refine] POST /command/core/get-importing-job-status (1002ms)
09:57:40.841 [                   refine] POST /command/core/get-importing-job-status (1002ms)
09:57:41.843 [                   refine] POST /command/core/get-importing-job-status (1002ms)
09:57:42.845 [                   refine] POST /command/core/get-importing-job-status (1001ms)
09:57:43.848 [                   refine] POST /command/core/get-importing-job-status (1002ms)
09:57:44.851 [                   refine] POST /command/core/get-importing-job-status (1002ms)
09:57:45.853 [                   refine] POST /command/core/get-importing-job-status (1002ms)
09:57:46.855 [                   refine] POST /command/core/get-importing-job-status (1002ms)
09:57:47.858 [                   refine] POST /command/core/get-importing-job-status (1002ms)
09:57:48.877 [                   refine] POST /command/core/get-importing-job-status (1019ms)
09:57:49.880 [                   refine] POST /command/core/get-importing-job-status (1002ms)
09:57:50.892 [                   refine] POST /command/core/get-importing-job-status (1012ms)
09:57:51.892 [                   refine] POST /command/core/get-importing-job-status (999ms)
09:57:52.896 [                   refine] POST /command/core/get-importing-job-status (1004ms)
09:57:53.897 [                   refine] POST /command/core/get-importing-job-status (1000ms)
09:57:54.897 [                   refine] POST /command/core/get-importing-job-status (1000ms)
09:57:55.903 [                   refine] POST /command/core/get-importing-job-status (1005ms)
09:57:56.914 [                   refine] POST /command/core/get-importing-job-status (1010ms)
09:57:57.914 [                   refine] POST /command/core/get-importing-job-status (1000ms)
09:57:58.915 [                   refine] POST /command/core/get-importing-job-status (1000ms)
09:57:59.925 [                   refine] POST /command/core/get-importing-job-status (1010ms)
09:58:00.931 [                   refine] POST /command/core/get-importing-job-status (1005ms)
09:58:01.931 [                   refine] POST /command/core/get-importing-job-status (1000ms)
09:58:02.937 [                   refine] POST /command/core/get-importing-job-status (1005ms)
09:58:03.942 [                   refine] POST /command/core/get-importing-job-status (1005ms)
09:58:04.953 [                   refine] POST /command/core/get-importing-job-status (1010ms)
09:58:05.964 [                   refine] POST /command/core/get-importing-job-status (1011ms)
09:58:06.980 [                   refine] POST /command/core/get-importing-job-status (1015ms)
09:58:07.993 [                   refine] POST /command/core/get-importing-job-status (1012ms)
09:58:08.996 [                   refine] POST /command/core/get-importing-job-status (1002ms)
09:58:10.009 [                   refine] POST /command/core/get-importing-job-status (1013ms)
09:58:11.024 [                   refine] POST /command/core/get-importing-job-status (1015ms)
09:58:12.038 [                   refine] POST /command/core/get-importing-job-status (1014ms)
09:58:13.046 [                   refine] POST /command/core/get-importing-job-status (1007ms)
09:58:14.048 [                   refine] POST /command/core/get-importing-job-status (1002ms)
09:58:15.063 [                   refine] POST /command/core/get-importing-job-status (1014ms)
09:58:16.079 [                   refine] POST /command/core/get-importing-job-status (1016ms)
09:58:17.081 [                   refine] POST /command/core/get-importing-job-status (1001ms)
09:58:18.082 [                   refine] POST /command/core/get-importing-job-status (1001ms)
09:58:19.088 [                   refine] POST /command/core/get-importing-job-status (1005ms)
09:58:20.104 [                   refine] POST /command/core/get-importing-job-status (1015ms)
09:58:21.103 [                   refine] POST /command/core/get-importing-job-status (999ms)
09:58:22.106 [                   refine] POST /command/core/get-importing-job-status (1002ms)
09:58:23.108 [                   refine] POST /command/core/get-importing-job-status (1002ms)`

--- a/src/main/java/org/openrefine/extensions/commons/importer/FileFetcher.java
+++ b/src/main/java/org/openrefine/extensions/commons/importer/FileFetcher.java
@@ -43,7 +43,7 @@ public class FileFetcher implements Iterator<JsonNode>{
      */
     public void getCallResults(String category) throws IOException {
 
-        OkHttpClient client = new OkHttpClient.Builder().build();
+        OkHttpClient client = HttpClient.getClient();
         urlBase = HttpUrl.parse(apiUrl).newBuilder()
                 .addQueryParameter("action", "query")
                 .addQueryParameter("list", "categorymembers")

--- a/src/main/java/org/openrefine/extensions/commons/importer/HttpClient.java
+++ b/src/main/java/org/openrefine/extensions/commons/importer/HttpClient.java
@@ -1,0 +1,35 @@
+package org.openrefine.extensions.commons.importer;
+
+import java.io.IOException;
+
+import okhttp3.Interceptor;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+
+public class HttpClient {
+
+    private static final String USER_AGENT = "OpenRefine-Commons-Extension/0.1.* (https://github.com/OpenRefine/CommonsExtension)";
+
+    public static OkHttpClient getClient() {
+        OkHttpClient client = new OkHttpClient.Builder().addInterceptor(new UserAgentInterceptor(USER_AGENT)).build();
+        return client;
+    }
+
+    static class UserAgentInterceptor implements Interceptor {
+
+        private final String userAgent;
+
+        public UserAgentInterceptor(String userAgent) {
+            this.userAgent = userAgent;
+        }
+
+        @Override
+        public Response intercept(Chain chain) throws IOException {
+            return chain.proceed(chain.request().newBuilder()
+                    .header("User-Agent", userAgent)
+                    .build());
+        }
+    }
+    private static OkHttpClient client;
+}

--- a/src/main/java/org/openrefine/extensions/commons/importer/RelatedCategoryFetcher.java
+++ b/src/main/java/org/openrefine/extensions/commons/importer/RelatedCategoryFetcher.java
@@ -60,7 +60,7 @@ public class RelatedCategoryFetcher implements Iterator<FileRecord> {
         while (titlesIndex < fileRecordOriginal.size()) {
             titles += "|" + fileRecordOriginal.get(titlesIndex++).fileName;
         }
-        OkHttpClient client = new OkHttpClient.Builder().build();
+        OkHttpClient client = HttpClient.getClient();
         HttpUrl urlRelatedCategoriesBase = HttpUrl.parse(apiUrl).newBuilder()
                 .addQueryParameter("action", "query")
                 .addQueryParameter("prop", "categories")

